Audio list info

Zainrax opened the pull request.

Zainrax committed:
refactor(Chart.js/Analysis): upgrade chart.js to 3.6.2, fix undefined usage, == to ===

Clean up in prep for using more performant 3.6, with migration/clean up along the way

Zainrax committed:
Merge remote-tracking branch 'origin' into audio-list-info

Zainrax committed:
feat(scripts): add deno scripts to add user, device, and recordings

Primary reason for using deno is to remove the need for more modules and speed to compile. These are light scripts to call and can be found here https://github.com/denoland/deno. They integrate with the existing types and don't require additional tsc scripts to run.

Zainrax committed:
feat(Audio): migrate additionalMetadata to own columns & tracks, add audio tags & cacophony index graph

Using sequelize to migrate data for audio recordings in Recordings table, taking cacophony_index to own cacophonyIndex column, audio Analysis(morepork tags) to Track & tags, and added corresponding UI elements to the Recording List

Zainrax committed:
fix(tests/audio): change cacophony index & analysis properties in tests, change types

This changes aims to access the new location of the cacophony index and analysis for audio with some fixes around reporting & accessing the new cacophonyIndex

Zainrax committed:
style(browse): automatic lint

Commit:
filter false positive tracks from ui

Commit:
comment

Commit:
margine

Commit:
re add template

Zainrax committed:
refactor(migration): change use of sequelize models to use only SQL queries

Was able to access the Models in the Travis CI, so reverted to only SQL. Generally not a good idea to use the models in migrations anyhow.

Zainrax committed:
fix(migration): check for algorithm in DetailSnapshots

Commit:
try render frame

Commit:
lint fix

Zainrax committed:
fix(migration): use replacements in sequelize migration, add automatic to Track table

Adding automatic to Track table as audio will have the ability to have manual tracks being added. It would be useful therefore to be able to distinguish them in a single query.

Commit:
add filtering to recordings view

Commit:
ran prettier

Zainrax committed:
feat(Audio): add Track List UI, Spectrogram, Track Selection, fix migration

This is a rough implementation of parts of the Audio UI along side a fix for migration

Commit:
fix human disagree logic

Zainrax committed:
fix(migration): check for cacophonyIndex

Zainrax committed:
fix(migration): default value for liklihood

Ran into issue where somes Tags don't have a liklihood

Zainrax committed:
fix(migration): reverse boolean check, add default value

Commit:
fix master tags

Commit:
linted

Commit:
updated

Commit:
updated styles and fixed sticky

hardiesoft commented on api/migrations/20220110021448-cacophonyIndex-to-Recordings.js:
```diff
@@ -0,0 +1,261 @@
+                    VALUES (:id, :algorithmId, :data, NOW(), NOW())
+                    RETURNING "id";`,
+                    {
+                      transaction,
```
I think it might be good to unify the concept of the track "geometry" between audio and thermal recordings.  I see that right now you're using the start and end time values to draw your boxes, but I think it would be good to be doing that from a rectangle structure like the `ApiTrackPosition` interface.  These numbers could just be normalised in the range 0..1, with y being 0 and height being 1 initially - so would draw the same boxes you're drawing now, but when we have frequency information we want to include, it wouldn't require any structural changes.  So you'd have `positions: [{ x: 0.25, width: 0.1, y: 0, height: 1 }]` with just a single "frame" of position info.  Thoughts?  We could also just have a different JSON structure here, and not try to shoehorn it into the same format as the thermal recordings - but it would be good to have the frequency bounds in there from day one.  Finally, do you have any more thoughts on how we can merge calls that are almost certainly from the same bird into a single "track"?  For the morepork tracks that you're creating in this migration, we might want to have a post-processing pass in the nearish future that attempts to merge a bunch of these "call" tracks into a single continuous track?

Zainrax replied:
Unifying values across thermal and audio where possible I think is super important, I'm still familiarising myself with the thermal side of things so guidance is always welcome. I simply took the current values and moved them over, but I'll normalise to the duration instead.
I think using normalised values is a much better idea in general, just for frequency we would either have to have to set a field for the max/min frequency at the time of making the track, or determine a static max/min frequency across all future tracks (which currently can't be set with the plugin spectrogram I'm using)
I've thought about merging (purposely put a partial track in the figma design to think about that exact case). There are either automatic, or manual tracks, so I think the simplest solution would be the ability to allow users to delete/change the location of an automatic track by dragging the corners to the right location and simply labelling it as a manual track. The other way  is to have logic around if box1 is close to box2 prompt the user to merge etc. which adds extra steps. So not sure what the nicest user experience is yet.
There will likely need post processing on the morepork inference, and even manual tracks, as consistent data typically builds better models.

hardiesoft commented on api/migrations/20220110021448-cacophonyIndex-to-Recordings.js:
```diff
@@ -0,0 +1,261 @@
+                  "Creating TrackTag for track",
+                  trackId,
+                  species,
+                  liklihood
```
liklihood -> likelihood

hardiesoft commented on api/migrations/20220110021448-cacophonyIndex-to-Recordings.js:
```diff
@@ -0,0 +1,261 @@
+      let algorithmId = await queryInterface.sequelize.query(`
+        SELECT "id" from "DetailSnapshots" 
+        WHERE "type"='algorithm' AND "details"->>'algorithm'='sliding_window';`);
+      if (!algorithmId[0].length !== 0) {
```
This if clause looks super dodgey...

Zainrax replied:
agreed

Commit:
add filtered column

Commit:
fix required

Commit:
update filtered

Zainrax committed:
feat(api/audio): add tagging to spectrogram, track api changes

This includes a several changes to the UI to allow for tagging functionality, and the ability to add and delete tracks. There are additional api ends including ability to get a single track which is more useful on the audio side where you make only have a few tracks. As well as including making positions for tracks normalized and include generic language (frame_num -> order).

Commit:
fixed up some tests

Zainrax committed:
fix(merge/test): merge with main, add automatic to expected tracks

Zainrax committed:
fix(lint): recording lint

Zainrax committed:
fix(recordings): register cacophony index graph

Zainrax committed:
Merge remote-tracking branch 'origin/main' into audio-list-info

Zainrax committed:
fix(merge): vue reactivity, api endpoints fix, add tracks/tags recording list

Several issues occured with merge, included fixes as well as reimplemententing the track & tags

Zainrax committed:
fix(audio): add track data type to track request, basic confirm

hardiesoft committed:
- Remove DeviceUsers from api and ui.
- Delete a whole lot of dead code from the api.

Zainrax committed:
fix(audio): allow backwards tracks, fix delete, add float to track pos

Zainrax committed:
refactor(scripts): remove personal scripts

gw7nvw committed:
progressing deviceuser removal from tests

gw7nvw committed:
all device users removed from tests

gw7nvw committed:
deleted test suites for endpoints we no longer support

hardiesoft committed:
- Refactoring recordings query

hardiesoft committed:
Merge branch 'migrate_device_users-tests' of https://github.com/TheCacophonyProject/cacophony-web into migrate_device_users

hardiesoft committed:
- Fix monitoring query

Zainrax committed:
feat(Vue): add composition-api, vue 2 compatability

This is a future proofing changing in case of moving to Vue 3. This way any new components can be made in a way that lessens migration issues. It also allows the use of Volar vue tooling.

Commit:
dont skip the test

hardiesoft committed:
- Fixing monitoring tests

hardiesoft committed:
- Fixing monitoring tests

hardiesoft committed:
- Fix duplicate visits timing issue.

hardiesoft committed:
- Fix group location map link on groupsview.

Commit:
Merge remote-tracking branch 'origin/main' into filter-tracks

Commit:
clean up merge

Commit:
Merge remote-tracking branch 'origin/migrate_device_users' into filter-tracks

Commit:
finished merging

Commit:
add back in additional meta

Commit:
add filtered

Commit:
update to work with separate tracks

Commit:
update filtered logic when no ai tags

Commit:
skip travis for now

Commit:
update api doc

Zainrax committed:
feat(audio): add immer, change colours, add mute icon, major refactor audio, typescript changes

This commit includes a number of larger changes including the use of composition api and immer along with styling and functionality changes on audio side. The reason for using composition is to support much better typing, with Volar also allowing for future Vue 3 use. Immer to implement immutable objects which allow for state management across time which is particularly useful to allow for redo/undo, which audio has more of (add/deleting/modifying tracks, along with the normal tags)

Commit:
fix no tags filtering and reprocess filtering

Commit:
set filtered on reprocess

Commit:
use method

Commit:
update filtered with transaction

Zainrax committed:
refactor(audio): functions to utils, add immutable state, style changes

This adds a useState to modify the reactive state of a ref with a proper immutable structure

gw7nvw committed:
Add filter tests, rework to use data from commands/dataTemplates

gw7nvw committed:
Add filter tests, rework to use data from commands/dataTemplates - bugfixes

gw7nvw committed:
Add filter tests, rework to use data from commands/dataTemplates - linting

Commit:
run tests, fix empty track not being filtered

Commit:
update filtered from fileprocessing api

Commit:
fixed lock file

Commit:
capital M for master on tests

Commit:
fix some tests

Commit:
remove only

Commit:
filtered true

gw7nvw committed:
completed filtering tests

gw7nvw committed:
centralise filtered_tags definition

gw7nvw committed:
tests - templates - stop tests from modifying global test-data templates

Commit:
fix z index

Commit:
Merge remote-tracking branch 'origin/filter-tracks-tests3' into filter-tracks

Zainrax committed:
feat(audio): undo functionality on tracks

Reuses the existing track in memory to add the track and tags

Zainrax committed:
feat(audio): add bird labels, playlist w/ nav, delete, download, useRoute/r, styling

Zainrax committed:
fix(audio): delete unused, add frequency to track, fix colours, and common bird buttons

Zainrax committed:
Merge remote-tracking branch 'origin/filter-tracks-tests3' into audio-list-info

Zainrax committed:
fix(audio): add freq from merge, disregard temp track colour

Zainrax committed:
fix(test): change frame_number to order

Zainrax committed:
fix(test): change cacophony index for test

Zainrax committed:
fix(audio): add cacophony index (tooltips), alphabetical list, check touch events available

Zainrax committed:
fix(audio): remove check for interface

Zainrax committed:
fix(recordings): check type is available in row view

Zainrax committed:
fix(recording): change props key, remove type column

gw7nvw committed:
merge in combined intergration tests

gw7nvw committed:
Finalised tests to run against test server (linting)

Zainrax committed:
fix(recordings): change z-index on sidebar on smaller screen

Zainrax committed:
fix(migration): remove Promise.all due to memory heap
